### PR TITLE
#11425: Remove tt_lib from models

### LINKED ISSUE #11425: Replace import tt_lib in blazepose, bloom, bloom_old, convnet_mnist, deit
Tracking Issue : #11240 , Master Issue : #10532 

- [x] blazers
- [x] bloom
- [x] bloom_old
- [x] convnet_mnist
- [x] deit

Current status: Ready to merge

### PR DESCRIPTION
**Pull request by Aswinmcw**
Tracking Issue : #11425 

### Checklist
- [ ] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by ayerofieiev-tt — APPROVED** [review]

**Comment by KalaivaniMCW** [comment]
Run nightly FD and model perf once to check pipelines

File: models/experimental/bloom/tt/bloom_attention.py [hunk 1]
```diff
@@ -7,11 +7,9 @@
 from torch.nn import functional as F
 
 import ttnn
-import tt_lib
 import models.experimental.bloom.bloom_utils as bloom_utils
 import models.experimental.bloom.tt.bloom_merge_heads as bloom_merge_heads
-from tt_lib.fused_ops.softmax import softmax as tt_softmax
```
Comment (KalaivaniMCW) [inline models/experimental/bloom/tt/bloom_attention.py:13]: keep this for now

File: models/experimental/bloom/tt/bloom_attention.py [hunk 2]
```diff
@@ -7,11 +7,9 @@
...
-
+from ttnn.experimental.fused_ops.softmax import softmax as tt_softmax
```
Comment (KalaivaniMCW) [inline models/experimental/bloom/tt/bloom_attention.py:12]: remove this

File: models/experimental/bloom_old/tt/bloom_attention.py [hunk 3]
```diff
@@ -259,18 +259,18 @@ def forward(
 
         query_layer = query_layer.transpose(1, 2)
         query_layer = bloom_utils.torch2tt_tensor(query_layer, device)
-        reshaped_query_layer = ttm.tensor.reshape(query_layer, 1, batch_size * self.num_heads, q_length, self.head_dim)
+        reshaped_query_layer = ttnn.reshape(query_layer, 1, batch_size * self.num_heads, q_length, self.head_dim)
```
Comment (KalaivaniMCW) [inline models/experimental/bloom_old/tt/bloom_attention.py:262]: use experimental reshape. or will have to wait till ttnn.reshape goes to main.
the arg syntax used here is not for ttnn.reshape

File: models/experimental/bloom_old/tt/bloom_attention.py [hunk 4]
```diff
@@ -259,18 +259,18 @@ def forward(
...
 
         # key_layer = key_layer.permute(0, 2, 3, 1).reshape(batch_size * self.num_heads, self.head_dim, q_length)
         key_layer = key_layer.permute(0, 2, 3, 1)
 
         key_layer = bloom_utils.torch2tt_tensor(key_layer, device)
-        reshaped_key_layer = ttm.tensor.reshape(key_layer, 1, batch_size * self.num_heads, self.head_dim, q_length)
+        reshaped_key_layer = ttnn.reshape(key_layer, 1, batch_size * self.num_heads, self.head_dim, q_length)
```
Comment (KalaivaniMCW) [inline models/experimental/bloom_old/tt/bloom_attention.py:268]: as above

File: models/experimental/bloom_old/tt/bloom_attention.py [hunk 5]
```diff
@@ -259,18 +259,18 @@ def forward(
...
 
         # value_layer = value_layer.transpose(1, 2).reshape(batch_size * self.num_heads, q_length, self.head_dim)
         value_layer = value_layer.transpose(1, 2)
         value_layer = bloom_utils.torch2tt_tensor(value_layer, device)
-        reshaped_value_layer = ttm.tensor.reshape(value_layer, 1, batch_size * self.num_heads, q_length, self.head_dim)
+        reshaped_value_layer = ttnn.reshape(value_layer, 1, batch_size * self.num_heads, q_length, self.head_dim)
```
Comment (KalaivaniMCW) [inline models/experimental/bloom_old/tt/bloom_attention.py:273]: as above

File: models/experimental/bloom_old/tt/bloom_attention.py [hunk 6]
```diff
@@ -284,7 +284,7 @@ def forward(
         )
 
         # change view to [batch_size, num_heads, q_length, kv_length]
-        attention_scores = ttm.tensor.reshape(matmul_result, batch_size, self.num_heads, q_length, kv_length)
+        attention_scores = ttnn.reshape(matmul_result, batch_size, self.num_heads, q_length, kv_length)
```
Comment (KalaivaniMCW) [inline models/experimental/bloom_old/tt/bloom_attention.py:287]: as above

File: models/experimental/bloom_old/tt/bloom_attention_merge_heads.py [hunk 7]
```diff
@@ -21,24 +19,20 @@ def merge_heads(
...
-) -> torch.Tensor:
+def tt_merge_heads(x: torch.Tensor, num_heads, hidden_size, num_attention_heads, device) -> torch.Tensor:
     head_dim = hidden_size // num_attention_heads
 
     batch_size_and_num_heads, seq_length, _ = x.shape
     batch_size = batch_size_and_num_heads // num_heads
 
     tt_x = bloom_utils.torch2tt_tensor(x, device)
-    tt_reshaped = ttm.tensor.reshape(tt_x, batch_size, num_heads, seq_length, head_dim)
+    tt_reshaped = ttnn.reshape(tt_x, batch_size, num_heads, seq_length, head_dim)
```
Comment (KalaivaniMCW) [inline models/experimental/bloom_old/tt/bloom_attention_merge_heads.py:29]: as above

File: models/experimental/bloom_old/tt/bloom_attention_merge_heads.py [hunk 8]
```diff
@@ -21,24 +19,20 @@ def merge_heads(
...
 
     p_reshaped = bloom_utils.tt2torch_tensor(tt_reshaped)
 
     # batch_size, num_heads, seq_length, head_dim -> batch_size, seq_length, num_heads, head_dim
     p_permuted = p_reshaped.permute(0, 2, 1, 3)
     tt_permuted = bloom_utils.torch2tt_tensor(p_permuted, device)
-    reshaped_2 = ttm.tensor.reshape(
-        tt_permuted, 1, batch_size, seq_length, num_heads * head_dim
-    )
+    reshaped_2 = ttnn.reshape(tt_permuted, 1, batch_size, seq_length, num_heads * head_dim)
```
Comment (KalaivaniMCW) [inline models/experimental/bloom_old/tt/bloom_attention_merge_heads.py:36]: as above

**Comment by KalaivaniMCW** [comment]
Left some comments about reshape. Also check with TT if they have any particular CI for this model or see if it runs on local.

**Review by KalaivaniMCW — APPROVED** [review]
Approving under the impression that review comments will be addressed before merge.

**Review by eyonland — APPROVED** [review]

**Pull request merged**